lib/src/map/controller/events/map_event_source.dart

```diff
@@ -0,0 +1,29 @@
+/// Event sources which are used to identify different types of
+/// [MapEvent] events
+enum MapEventSource {
```

Comment:
I agree that the event system should get revised. But in my opinion the "event system" and "gesture system" are two different things therefore I'd prefer to have those changes not bundled in this pull request.
My main point is that some MapEventSources have their own dedicated Event class which should be avoided. But the specification should be not discussed here because off topic. We can however delay this pull request until we have reworked the gesture system and integrate it here.

Reply:
> in addition I really like to get rid of the event id property as they don't get used by gestures anyway.

It can be passed through from the `MapController` (or at least it should be if it can't atm), and is very useful in some cases.

> But in my opinion the "event system" and "gesture system" are two different things therefore I'd prefer to have those changes not bundled in this pull request.

I'm not so sure that they're that unrelated, but we can keep them seperate. We just really need to revisit it, as that is also a weak point in the entire gesture area.

Reply:
> It can be passed through from the MapController (or at least it should be if it can't atm), and is very useful in some cases.

Would it make sense to change the id type to `Object`? That way users would input everything they like and don't have to format and parse Strings.